scalyr_agent/builtin_monitors/kubernetes_events_monitor.py

```diff
@@ -583,87 +583,99 @@ def run(self):
+                            if kind:
+                                if kind == 'Pod':
+                                    extra_fields['pod_name'] = name
+                                    extra_fields['pod_namespace'] = namespace
+                                    pod = k8s_cache.pod(namespace, name, current_time,
+                                                        query_options=k8s_events_query_options)
+                                    if pod and pod.controller:
+                                        extra_fields['k8s-controller'] = pod.controller.name
+                                        extra_fields['k8s-kind'] = pod.controller.kind
+                                elif kind != 'Node':
+                                    controller = k8s_cache.controller(namespace, name, kind, current_time,
+                                                                      query_options=k8s_events_query_options)
+                                    if controller:
+                                        extra_fields['k8s-controller'] = controller.name
+                                        extra_fields['k8s-kind'] = controller.kind
+
+                            # if so, log to disk
+                            self.__disk_logger.info( 'event=%s extra=%s' % (str(json_lib.serialize(obj)),
+                                                                            str(json_lib.serialize(extra_fields))))
+
+                            # see if we need to check for a new leader
+                            if last_check + self._leader_check_interval <= current_time:
+                                global_log.log( scalyr_logging.DEBUG_LEVEL_1, "Time to check for a new event leader" )
+                                break
                     
-                except K8sApiAuthorizationException:
-                    global_log.warning("Could not stream K8s events due to an authorization error.  The "
-                                       "Scalyr Service Account does not have permission to watch available events.  "
-                                       "Please recreate the role with the latest definition which can be found "
-                                       "at https://raw.githubusercontent.com/scalyr/scalyr-agent-2/release/k8s/scalyr-service-account.yaml "
-                                       "K8s event collection will be disabled until this is resolved.  See the K8s install "
-                                       "directions for instructions on how to create the role "
-                                       "https://www.scalyr.com/help/install-agent-kubernetes", limit_once_per_x_secs=300,
-                                       limit_key='k8s-stream-events-no-permission')
-                except ConnectionError, e:
-                    # ignore these, and just carry on querying in the next iteration
+                        except K8sApiAuthorizationException:
```

Comment:
This can't belong inside the `for line in lines` loop, because if there is an authorization problem (i.e. a misconfigured service account) then the exception will be raised by line 576 `lines = k8s_api_events.stream_events( last_event=last_event )` which will now only print a generic exception message rather than a detailed message about how to fix authorization issues.

Do you mean to be catching other generic API errors here?

Reply:
Good catch . I'll relocate it back to original position and just keep a single generic exception catcher inside the loop.